crypto/pem/pem_pk8.c

```diff
@@ -93,7 +93,7 @@ static int do_pk8pkey(BIO *bp, const EVP_PKEY *x, int isder, int nid,
         }
     }
 
-    if (OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
+    if (nid == -1 && OSSL_ENCODER_CTX_get_num_encoders(ctx) != 0) {
```

Comment:
@levitte - is this right or should it be converting the nid?

Reply:
IMO it should convert the nid to EVP_CIPHER and only if fetch fails it should fallback to the legacy path.

Reply:
Started adding the code and then thought...
Wont that be kind of pointless since these would be NIDS related to pbe that are not fetched.
e.g: NID_pbe_WithSHA1And3_Key_TripleDES_CBC

Reply:
Code would also be ugly for a case that is not used...
````
    if (nid != -1 && enc == NULL) {
        OSSL_LIB_CTX *libctx = NULL;

        if (evp_pkey_is_provided(x)) {
            const OSSL_PROVIDER *prov = EVP_KEYMGMT_provider(x->keymgmt);

            libctx = ossl_provider_libctx(prov);
        }
        pbe_cipher = EVP_CIPHER_fetch(libctx, OBJ_nid2sn(nid), propq);
        enc = pbe_cipher;
    }
````

Reply:
Hmm, you're right. Unless we in future add a possibility to fetch PBEs it would be pointless. So perhaps just add a comment?

Reply:
Done.